### PR TITLE
Fix Gluon Language Model Example

### PR DESCRIPTION
**Pull request by szhengac**
## Description ##
The current language model example has bad parameter setting. This pr fixes this.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: example/gluon/word_language_model/train.py [hunk 1]
```diff
@@ -159,19 +160,19 @@ def train():
...
                 L.backward()
 
             grads = [p.grad(context) for p in model.collect_params().values()]
-            # Here gradient is for the whole batch.
-            # So we multiply max_norm by batch_size and bptt size to balance it.
-            gluon.utils.clip_global_norm(grads, args.clip * args.bptt * args.batch_size)
+            gluon.utils.clip_global_norm(grads, args.clip)
 
-            trainer.step(args.batch_size)
+            trainer.step(1)
```
Comment (ThomasDelteil) [inline example/gluon/word_language_model/train.py]: Why do you change the step to 1?

Comment (szha) [reply example/gluon/word_language_model/train.py]: The loss was rescaled manually above

Comment (szhengac) [reply example/gluon/word_language_model/train.py]: Yes, the loss has been rescaled manually. Also, we should rescale the loss by batch_size * bptt instead.

File: example/gluon/word_language_model/train.py [hunk 2]
```diff
@@ -188,11 +189,7 @@ def train():
             print('test loss %.2f, test ppl %.2f'%(test_L, math.exp(test_L)))
         else:
             args.lr = args.lr*0.25
-            trainer._init_optimizer('sgd',
-                                    {'learning_rate': args.lr,
-                                     'momentum': 0,
-                                     'wd': 0})
-            model.collect_params().load(args.save, context)
+            trainer.set_learning_rate(args.lr)
```
Comment (yongsun) [inline example/gluon/word_language_model/train.py]: The original logic will decay learning rate, and reload previous saved parameters to continue the training. Why the reloading is removed?

File: example/gluon/word_language_model/train.py [hunk 3]
```diff
@@ -27,23 +27,23 @@
...
-parser.add_argument('--emsize', type=int, default=200,
+parser.add_argument('--emsize', type=int, default=650,
                     help='size of word embeddings')
-parser.add_argument('--nhid', type=int, default=200,
+parser.add_argument('--nhid', type=int, default=650,
                     help='number of hidden units per layer')
 parser.add_argument('--nlayers', type=int, default=2,
                     help='number of layers')
-parser.add_argument('--lr', type=float, default=1.0,
+parser.add_argument('--lr', type=float, default=20,
```
Comment (yongsun) [inline example/gluon/word_language_model/train.py]: Why the default lr bumped to 20? Is it because it should be re-scaled by batch_size?

File: example/gluon/word_language_model/train.py (hunk 2 above)
Comment (szhengac) [reply example/gluon/word_language_model/train.py]: There is evidence that reloading previous saved parameter is better. Keeping the current parameter is instead theoretically supported in the view of optimization.

File: example/gluon/word_language_model/train.py (hunk 3 above)
Comment (szhengac) [reply example/gluon/word_language_model/train.py]: Yes, the gradient is now normalized.
